### PR TITLE
Babeltrace2 simple filter example

### PR DESCRIPTION
**Pull request by DonAurelio**

### CODE REVIEW

File: example_bt2/README.MD
Comment (TApplencourt) [inline example_bt2/README.MD:1]: You can put them in the example. 
And use the same name that they use (distil?)
You can use the same source as the example source.

File: example_bt2/params copy.yaml
Comment (TApplencourt) [inline example_bt2/params copy.yaml:1]: ????

File: example_bt2/params copy.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+:type: map
```
Comment (TApplencourt) [inline example_bt2/params copy.yaml:1]: What are they using in their example, string or array of string?

File: example_bt2/btx_filter/callbacks.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  strcpy(str, data->names_value);
+
+  bool ocurrence = false;
+
+  char *ptr = strtok(str, ",");
+  while(ptr != NULL && !ocurrence)
+  {
+    ocurrence = (strcmp(ptr,event_class_name) == 0);
+    ptr = strtok(NULL, ",");
+  }
```
Comment (TApplencourt) [inline example_bt2/btx_filter/callbacks.c:42]: We should addapt their code:
```
for (uint64_t i = 0; i < [bt_value_array_get_length](https://babeltrace.org/docs/v2.0/libbabeltrace2/group__api-val.html#gaf46c04faf88e6ebd44db4ba40b3b3fe6)(
            distill_iter->distill->names_value); i++) {
        const char *discard_name = [bt_value_string_get](https://babeltrace.org/docs/v2.0/libbabeltrace2/group__api-val.html#ga7e98e940d278a5b7e9c05081d25c1e28)(
            [bt_value_array_borrow_element_by_index_const](https://babeltrace.org/docs/v2.0/libbabeltrace2/group__api-val.html#ga534d3ba2056438eb0181f4a978379517)(
                distill_iter->distill->names_value, i));
 
        if (strcmp(name, discard_name) == 0) {
            passes = false;
            goto end;
        }
    }
```

The goal is to stay as close as possible as their example.

File: example_bt2/btx_filter/callbacks.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  // If the event_class_name is found in 'data->names_value', the associated callback is called,
+  // this resulting in the event being discared.
+  *matched = ocurrence;
+}
+
+static void btx_filter_callback(void *btx_handle, void *usr_data, const char* event_class_name)
+{
+
```
Comment (TApplencourt) [inline example_bt2/btx_filter/callbacks.c:51]: I don't think we need an empty `{}` here...

File: example_bt2/btx_source/callbacks.c
Comment (TApplencourt) [inline example_bt2/btx_source/callbacks.c:1]: we can reuse the other source in the example

File: example_bt2/params copy.yaml (hunk 1 above)
Comment (DonAurelio) [reply example_bt2/params copy.yaml:1]: They use array of strings, but we do not support [that] (https://github.com/TApplencourt/metababel/blob/45878ae43e653a53764a7b5b401678821fb98d97/lib/metababel/bt2_values_generator.rb#L11), so I used string.

File: example_bt2/btx_filter/callbacks.c (hunk 2 above)
Comment (DonAurelio) [reply example_bt2/btx_filter/callbacks.c:42]: They use array of strings, but we do not support that, so I used string. I can convert a [comma separed string](https://github.com/portfoliocourses/c-example-code/blob/main/split.c) to an array, then replicate their code. But this add more lines of code. What do you think?

File: example_bt2/btx_filter/callbacks.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+typedef struct data_s data_t;
+
+void btx_initialize_usr_data(void *btx_handle, void **usr_data) {
+  *usr_data = calloc(1, sizeof(data_t));
+}
+
+void btx_read_params(void *btx_handle, void *usr_data, btx_params_t *usr_params) {
+  ((data_t *)usr_data)->names_value = usr_params->names;
```
[the remark this answers is not in the document]
Comment (DonAurelio) [reply example_bt2/btx_filter/callbacks.c:18]: Yes, we doble are safe !!

https://github.com/TApplencourt/metababel/blob/45878ae43e653a53764a7b5b401678821fb98d97/template/filter.c.erb#L411

```bash
==11627== 
==11627== HEAP SUMMARY:
==11627==     in use at exit: 24,170 bytes in 14 blocks
==11627==   total heap usage: 1,558 allocs, 1,544 frees, 390,112 bytes allocated
==11627== 
==11627== LEAK SUMMARY:
==11627==    definitely lost: 50 bytes in 1 blocks
==11627==    indirectly lost: 0 bytes in 0 blocks
==11627==      possibly lost: 0 bytes in 0 blocks
==11627==    still reachable: 24,120 bytes in 13 blocks
==11627==         suppressed: 0 bytes in 0 blocks
==11627== Rerun with --leak-check=full to see details of leaked memory
==11627== 
==11627== For lists of detected and suppressed errors, rerun with: -s
==11627== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

File: example_bt2/btx_filter/callbacks.c (hunk 3 above)
Comment (DonAurelio) [reply example_bt2/btx_filter/callbacks.c:51]: Good point !!!

File: example_bt2/params copy.yaml
Comment (DonAurelio) [reply example_bt2/params copy.yaml:1]: My bad

File: example_bt2/btx_source/callbacks.c
Comment (DonAurelio) [reply example_bt2/btx_source/callbacks.c:1]: Do you mean this?

https://github.com/TApplencourt/metababel/blob/45878ae43e653a53764a7b5b401678821fb98d97/example/btx_source/callbacks.c#L9

I tried to mimic the messages they are trying to discard. But the source above does not have that.
`--params='names=["sched_switch", "rcu_utilization", "kmem_kfree"]'`

Comment (TApplencourt) [reply example_bt2/btx_source/callbacks.c:1]: No I mean this source `https://github.com/TApplencourt/metababel/tree/main/example/btx_dust`, we should remove the other one.

Comment (TApplencourt) [reply example_bt2/btx_source/callbacks.c:1]: We can add some new dust imput with `sched_switch` for example in the dust

File: example/btx_dust/dust.txt [hunk 5]
```diff
@@ -3,3 +3,6 @@
 1578694241 001831 send-msg Bacon ipsum dolor amet strip steak.
 1578694241 944187 send-msg Spare ribs filet mignon boudin bresaola.
 1578694245 115406 recv-msg Rump cow t-bone hamburger short tenderloin.
+1578694246 001831 sched_switch Bacon ipsum dolor amet strip steak.
+1578694247 944187 rcu_utilization Spare ribs filet mignon boudin bresaola.
+1578694248 115406 kmem_kfree Rump cow t-bone hamburger short tenderloin.
```
Comment (TApplencourt) [inline example/btx_dust/dust.txt:8]: Please change the string :) If not it's a little conusing

**Pull request merged**